cmake/podioMacros.cmake

```diff
@@ -5,100 +5,22 @@
 function(PODIO_GENERATE_DICTIONARY dictionary)
-  CMAKE_PARSE_ARGUMENTS(ARG "" "SELECTION" "OPTIONS;DEPENDS" ${ARGN})
-  #---Get List of header files---------------
-  set(headerfiles)
-  foreach(fp ${ARG_UNPARSED_ARGUMENTS})
-    file(GLOB files inc/${fp})
-    if(files)
-      foreach(f ${files})
-        if(NOT f MATCHES LinkDef)
-          set(headerfiles ${headerfiles} ${f})
-        endif()
-      endforeach()
-    elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${fp})
-      set(headerfiles ${headerfiles} ${CMAKE_CURRENT_SOURCE_DIR}/${fp})
-    else()
-      set(headerfiles ${headerfiles} ${fp})
-    endif()
-  endforeach()
-  #---Get Selection file------------------------------------
-  if(IS_ABSOLUTE ${ARG_SELECTION})
-    set(selectionfile ${ARG_SELECTION})
-  else()
-    set(selectionfile ${CMAKE_CURRENT_SOURCE_DIR}/${ARG_SELECTION})
-  endif()
-
-  set(gensrcdict ${dictionary}.cxx)
+  REFLEX_GENERATE_DICTIONARY(${ARGV})
 
-  #---roottest compability---------------------------------
-  if(CMAKE_ROOTTEST_NOROOTMAP)
-    set(rootmapname )
-    set(rootmapopts )
-  elseif(DEFINED CMAKE_ROOTTEST_NOROOTMAP)  # Follow the roottest dictionary library naming
-    set(rootmapname ${dictionary}.rootmap)
-    set(rootmapopts --rootmap=${rootmapname} --rootmap-lib=${libprefix}${dictionary}_dictrflx)
-  else()
-    set(rootmapname ${dictionary}Dict.rootmap)
-    set(rootmapopts --rootmap=${rootmapname} --rootmap-lib=${libprefix}${dictionary}Dict)
-  endif()
-
-  set(include_dirs ${CMAKE_CURRENT_SOURCE_DIR})
-  get_directory_property(incdirs INCLUDE_DIRECTORIES)
-  foreach(d ${incdirs})
-    if(NOT "${d}" MATCHES "^(AFTER|BEFORE|INTERFACE|PRIVATE|PUBLIC|SYSTEM)$")
-      list(APPEND include_dirs ${d})
-    endif()
-  endforeach()
-
-  get_directory_property(defs COMPILE_DEFINITIONS)
-  foreach( d ${defs})
-   list(APPEND definitions ${d})
-  endforeach()
-
-  IF(TARGET ${dictionary})
-    LIST(APPEND include_dirs $<TARGET_PROPERTY:${dictionary},INCLUDE_DIRECTORIES>)
-    LIST(APPEND definitions $<TARGET_PROPERTY:${dictionary},COMPILE_DEFINITIONS>)
-  ENDIF()
-
-  add_custom_command(
-    OUTPUT ${gensrcdict} ${rootmapname}
-    COMMAND ${ROOT_genreflex_CMD}
-    ARGS ${headerfiles} -o ${gensrcdict} ${rootmapopts} --select=${selectionfile}
-         --gccxmlpath=${GCCXML_home}/bin ${ARG_OPTIONS}
-         "-I$<JOIN:${include_dirs},;-I>"
-         "$<$<BOOL:$<JOIN:${definitions},>>:-D$<JOIN:${definitions},;-D>>"
-    DEPENDS ${headerfiles} ${selectionfile} ${ARG_DEPENDS}
-
-    COMMAND_EXPAND_LISTS
-    )
-  IF(TARGET ${dictionary})
-    target_sources(${dictionary} PRIVATE ${gensrcdict})
-  ENDIF()
-  set(gensrcdict ${dictionary}.cxx PARENT_SCOPE)
+  # Dictionary payloads become very large for our generated EDMs, so we
+  # explicitly silence this warning here.
   set_source_files_properties(${gensrcdict}
     PROPERTIES
     GENERATED TRUE
     COMPILE_FLAGS "-Wno-overlength-strings"
```

Comment:
Maybe it's not great that the length is 10 times what compilers are required to support, maybe that's a problem some day...